modules/steam.nix

```diff
@@ -247,6 +247,39 @@ in
           '';
         };
 
+        autoStart = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to automatically launch the Steam Deck UI on boot.
+
+            No Display Managers may be enabled in conjunction with this option.
+          '';
+        };
+
+        user = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+            The user to run Steam with.
+
+            This user must be in the `networkmanager` group to be able to
+            complete the OOBE (Out-of-Box Experience).
+          '';
+        };
+
+        desktopSession = mkOption {
+          type = types.str;
+          default = "steam-wayland";
+          example = "plasma";
+          description = ''
+            The session to launch for Desktop Mode.
+
+            By default, attempting to switch to the desktop will launch
+            Gaming Mode again.
```

Comment:
Ouch. This is not great UX. I get that we don't want to force Plasma on people but also ouch.

Reply:
Maybe we can have a sentinel default value where it will be set to `steam-wayland` when the default is kept, *and also* warn, telling the user to set it to `steam-wayland` themselve if it is the desired outcome?

Reply:
Oof, I forgot to apply this option in the refactor. The Deck UI hardcodes `steamos-session-select plasma` to switch to the desktop, and we were rewriting it to the user's preferred session.

Reply:
You realized a mistake on your end, but I don't think this is what @K900 was commenting about, and that wasn't what I was commenting about.

I think the (slight) problem here is UX and defaults: it would be good if there was some way to warn the user that the default is "surprising", in that it will be basically a no-op.

Reply:
Missed this one - Now a warning is emitted if `desktopSession` is unset.